### DISCUSSION TITLE
undefined reference to `pugi::xml_document::xml_document()'

### DISCUSSION THREAD

**Discussion by snowyowl0428** [zeux/pugixml#521] [Q&A] [answered]
I used  pugi  to initialize an instance, but report a problem:
/home/ubuntu/pugixml/test.cpp:6: undefined reference to `pugi::xml_document::xml_document()'
/usr/bin/ld: /home/ubuntu/pugixml/test.cpp:6: undefined reference to `pugi::xml_document::~xml_document()'
it looks like a link problem but I have no idea how to fix it out.
Any help is appreciated.

**Comment by zeux** [comment, marked as answer]
You need to link to pugixml library somehow, either by using CMake dependency, or by adding pugixml.cpp to your compilation project, or by depending on a system-provided pugixml library (libpugixml-dev might install it on Ubuntu). See https://pugixml.org/docs/manual.html#install.building
